packages/admin/admin/src/stack/Stack.tsx

```diff
@@ -151,6 +141,15 @@ export function Stack(props: StackProps) {
         });
     }, []);
 
+    React.useEffect(() => {
+        // execute on location change, set stateUrl for the last breadcrumb to the current location
+        setBreadcrumbs((old) => {
+            const sorted = sortByParentId(old);
+            sorted[sorted.length - 1].stateUrl = location.pathname + location.search; // modify object in place
+            return [...old]; // clone (with modified object) to new array to trigger a state update
```

Comment:
Wouldn't it be better if sortByParentId wouldn't sort in place?

Reply:
I don't understand, sortByParentId does not sort in place?

Reply:
Why do you need to clone then?

Reply:
To trigger a state update.

That's how react works. You can't modify a deep nested object, you always have to create a new object at the root, so react knows the state changed. I didn't test tough if that is really needed in that case, basically I just ported the old code.

Reply:
Now I see: You're returning `[...old]`, not `sorted`. Is this even correct?

Reply:
the state variable itself is not stored sorted, it's sorted when showing the breadcrumbs. And the new feature needs to change the url of the last visible breadcrumb, that's why I also sort it here.

We could change that, that would make this particular function a bit simpler.

Reply:
My main concern is that this setState update isn't pure, since you're modifying the object. Would be interesting if this leads to an error in React StrictMode, which behaves a little bit different during development. But it's possible fine this way.

https://github.com/facebook/react/issues/12856

Reply:
I changed that now, the state variable is now sorted